src/transformers/cache_utils.py

```diff
@@ -2014,10 +2015,12 @@ def reset(self):
 
 class MambaCache:
     """
-    Cache for mamba model which does not have attention mechanism and key value states.
+    Cache for mamba model which does not have attention mechanism and key value states.  At initialization, the cache
+    is preallocated to its maximum possible shape. Contrarily to other caches, `max_batch_size` must match the
+    batch size used at inference time.
```

Comment:
Note: adding the feature to `MambaCache` would imply adding model-level changes. `mamba` doesn't have a lot of usage to justify the work